sklearn/ensemble/gradient_boosting.py

```diff
@@ -1334,10 +1333,11 @@ def _check_params(self):
                              "integer. %r was passed"
                              % self.n_iter_no_change)
 
-        allowed_presort = ('auto', True, False)
-        if self.presort not in allowed_presort:
-            raise ValueError("'presort' should be in {}. Got {!r} instead."
-                             .format(allowed_presort, self.presort))
+        if self.presort != 'deprecated':
+            warnings.warn("The parameter 'presort' is deprecated and has no "
+                          "effect. It will be removed in v0.24. You can "
+                          "suppress this warning by not passing any value "
+                          "to the 'presort' parameter.", UserWarning)
```

Comment:
Should this be a deprecation warning?

Also, don't we want to maintain backward compatibility?

Reply:
Since `presort` do not change the results but only affect speed, we can directly remove the effect. It will just be slower for some case (shallow max_depth). However, we will gain in readibility and we can probably parallelize the code with a prange and it will be a winner

Reply:
we are, not sure what you mean by maintaining backward compatibility here.

Reply:
Right sorry (still early here ^^)

I would prefer forcing to pass `False` to disable the warning though. Anything else should raise the warning. Users whose code is `presort=True` won't see the deprecation warning with the current design.

Reply:
Here we make sure that the warning is always raised, and the way to suppress it to not give any input, which is handy when we remove the parameter:

```python
@pytest.mark.parametrize('Cls',
                         (DecisionTreeRegressor, DecisionTreeClassifier))
@pytest.mark.parametrize('presort', ['auto', True, False])
def test_presort_deprecated(Cls, presort):
    X = np.zeros((10, 10))
    y = np.r_[[0] * 5, [1] * 5]
    tree = Cls(presort=presort)
    with pytest.warns(DeprecationWarning,
                      match="The parameter 'presort' is deprecated "):
        tree.fit(X, y)
```